python/cudf/cudf/dataframe/dataframe.py

```diff
@@ -2204,7 +2204,8 @@ def set_categories(col, cats):
                 cats = rcats
                 lhs[idx_col_name] = set_categories(lhs[idx_col_name], cats)
             elif how in ["inner", "outer"]:
-                cats = columnops.as_column(lcats).append(rcats).unique()
+                cats = columnops.as_column(lcats).append(rcats)
+                cats = Series(cats).drop_duplicates()._column
```

Comment:
Is the reason for using a Series instead of a column here that accessing `drop_duplicates` at the column level is problematic?

Reply:
Yeah, `drop_duplicates` preserves the order of the original categories, but in testing I noticed unique didn't. Preserving the original (first-seen) category order is necessary to match pandas `as_ordered()` behavior.